### ISSUE TITLE
Code blocks not rendering in Assistant response

### ISSUE THREAD

**Issue by omerbensaadon** [zed-industries/zed#20724]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

![Screenshot 2024-11-15 at 8 59 21 AM](https://github.com/user-attachments/assets/322d5030-7b25-46dc-b114-6e69ccdfd8b5)

In the Assistant pane, using Claude Sonnet 3.5, Code Blocks are not rendering? Not sure if this is a bug or a feature request

### Environment

Zed: v0.161.2 (Zed)
OS: macOS 14.6.1
Memory: 32 GiB
Architecture: aarch64

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary>

```

```
</details>

**Comment by notpeter** [comment]
Everything in the Assistant panel is presented as Plaintext/Markdown unmodified so that you can edit/manipulate as desired.

Can you be specific what you are expecting to happen?

**Comment by omerbensaadon** [comment]
hey! yeah I am expecting these to show up as code blocks like so: 

```
Code!
```

Ideally with language-specific syntax highlighting so: 

```python
import pandas
```

**Comment by Seltonu** [comment]
Seconding this. Having assistant responses respect code blocks makes the chat far more readable imo 

To better explain the need, here is an example where having code blocks would be very appreciated

![screenshot-2024-11-20-16-54-07](https://github.com/user-attachments/assets/b9871b37-3cb0-419a-8949-498c2926df15)

vs. the sample prompt on Claude web:

![screenshot-2024-11-20-17-07-01](https://github.com/user-attachments/assets/f857b1a5-93db-4fab-b96b-4fdf7e511419)

Note because the assistant panel response includes small snippets of PHP code with no `<?php ?>` tags present it also doesn't know to render syntax highlighting, compounding the problem. For PHP responses its often a tough wall of text to parse through.

**Comment by notpeter** [comment]
Gotcha.  This is a php-specific bug. 
Other languages render properly for me, but PHP is broken:
<img width="615" alt="Screenshot 2024-11-22 at 15 29 22" src="https://github.com/user-attachments/assets/5abe5fe3-446c-4a9c-8484-7553d57c9c53">

We use [tree-sitter/tree-sitter-php](https://github.com/tree-sitter/tree-sitter-php/) to handle our syntax highlighting for PHP and it appears that this tree-sitter does not correctly highlight snippets.  You can see this by pasting that same content into a Zed buffer and selecting PHP from the language selector.  It will not highlight correctly, but if you put a `<?php` and `?>` on the first/last lines everything works properly.

Zed will attempt syntax highlighting on markdown code blocks in the Assistant for any language it supports (including those provided by extensions).  So for example Shell Scripts or command lines can be tagged as `bash` or `sh` and they will show up as expected.

<img width="700" alt="Screenshot 2024-11-22 at 15 35 32" src="https://github.com/user-attachments/assets/e234d56a-e957-425c-adc0-0a6b72186fe3">

If you notice any other languages not being handled correctly please open issues.
Thanks for reporting!

**Comment by Seltonu** [comment]
FWIW aside from the PHP syntax highlighting bug, I still agree with OP that the code blocks themselves should ideally be rendered out for visual clarity. A dark box around code is much neater/more visually obvious separation in long assistant responses 🙂

**Comment by 88potatoes** [comment]
> Everything in the Assistant panel is presented as Plaintext/Markdown unmodified so that you can edit/manipulate as desired.
> 
> Can you be specific what you are expecting to happen?

Is it possible to make it a setting: switch from plaintext to code blocks?

**Comment by notpeter** [comment]
> Is it possible to make it a setting: switch from plaintext to code blocks?

There is active development work to restyle the Assistant Panel, including support for displaying code blocks not just with a markdown `` ```python ... ``` `` style fence but with a more polished code block widget. Nothing to announce yet, but just wanted to let you know that support for this is coming.

**Comment by adhorodyski** [comment]
Was just looking for exactly that, excited some changes are coming! Can't wait to try it out.

**Comment by notpeter** [comment]
> Was just looking for exactly that, excited some changes are coming! Can't wait to try it out.

Beta waitlist signup is here: https://zed.dev/ai/agent

**Comment by Seltonu** [comment]
Syntax highlighting of snippets remains unfixed for PHP where the `<?php ?>` tags are missing. As mentioned this is an issue with [tree-sitter/tree-sitter-php](https://github.com/tree-sitter/tree-sitter-php/)


<img width="501" height="427" alt="Image" src="https://github.com/user-attachments/assets/d807c199-6974-4660-acfc-3b1bffec6e3b" />

**Comment by notpeter** [comment]
Filed an issue upstream:
- https://github.com/tree-sitter/tree-sitter-php/issues/287

**Comment by Seltonu** [comment]
@omerbensaadon I dont believe this is complete, the lack of syntax highlighting is still present and the upstream issue @notpeter tagged suggested its an issue with Zed's configuration

<img width="388" height="906" alt="Image" src="https://github.com/user-attachments/assets/bd1d0589-3249-4737-a4d5-2adcbd6c4af5" />

**Comment by notpeter** [comment]
This is broken in current Zed (0.215.x). PHP code still does not syntax highlight in markdown (md files, markdown preview, agent/assistant code blocks) because it is using the `php` grammar instead of the `php_only` grammar.

To support this, I believe would require:
1. Add the `php_only` grammar and language to the Zed PHP extension.
  a. duplicate [these lines in extension.toml](https://github.com/zed-extensions/php/blob/9a119b18eeb247072964a19ce46fab54bbd1bb30/extension.toml#L25-L28) switching `path = "php"` to `path = "php_only"`
  b. add languages/php_only/config.toml
  c. symlink or adapt languages/php/*.scm to languages/php_only/*.scm 
2. Add a wrapper for the `language_for_name_or_extension` function (e.g. `language_for_code_fence_block`) which checked a hard-coded grammar override table (e.g. php=>php_only)

**Comment by notpeter** [comment]
Still broken
